### DISCUSSION TITLE
How to delay configure / reset / update flags?

### DISCUSSION THREAD

**Discussion by miller-productions** [sergiodxa/flagged#545] [Q&A] [answered]
Heya, apologies if this is a basic (aka naive) question. 

I'm just thinking through a scenario.

Say you're using `flagged` and you want to delay the configuration of its flags until some event happens. For example, say you're in a scenario that requires users to login to use the system. So, when the user login completes you receive details from the server about which flags that this particular user has access to and you use this to configure / reset / update `flagged` in your app.

How would you do this?

I've played around with `context` as one possibility here. i.e. you set up a `context`, `flagged` is configured based on the value(s) in that `context` and then when the login happens you go ahead and update that `context` to the latest settings. 

Another consideration, you might also need to reset the flags configuration, i.e. if the browser is reloaded.

What do you think? Is my question valid here in this context and is it something you would do with `flagged`, or am I off track?

**Comment by sergiodxa** [comment, marked as answer]
Keep your list of flags somewhere (like in a state) and pass them to FlagsProvider. When they change your components depending on them will re-render and update to the new flags.
